wallet/src/repository/wallet/mod.rs

```diff
@@ -1317,6 +1311,26 @@ where
     ) -> Result<Option<model::BalanceMovement>> {
         let mut state = self.state.write()?;
 
+        // Mark UTXOs as used so we don't double spend
+        // Save the timestamp to after which the UTXO can be spent again
+        let tx_pending_timeout = u64::from(state.epoch_constants.checkpoints_period) * 10;
+        let timestamp = u64::try_from(get_timestamp()).unwrap();
+        let mut wallet_utxos = WalletUtxos {
+            utxo_set: &UtxoSet::default(),
```

Comment:
This is not needed because `wallet_utxos.set_used_output_pointer` only modifies the `used_outputs` field, but it looks like a hack so it's better to simply use a reference to the utxo set here

```suggestion
            utxo_set: &state.utxo_set,
```